### PR TITLE
Update DG - use case, developer testing, and model diagram

### PR DESCRIPTION
**Pull request by aexolate**
- Updates the DG use cases and developer testing for `import` and `export` commands.
- Updates the Model diagram to use the correct attributes of `Person` class. 

Also, should the filtered label in the diagram be changed to filtered & sorted?

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -761,3 +805,37 @@ testers are expected to do more *exploratory* testing.
 
     1. Test case: `history 0`<br>
        Expected: Error details shown in the status message.
+
+### Exporting data file
+
+1. Exporting the data file.
+
+    1. Test case: `export fy2324.json`
```
Comment (Goh-Li-Ting) [inline docs/DeveloperGuide.md:813]: You can add `<br>` here and for the importing section to make it neater.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -761,3 +805,37 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case: `export fy2324`<br>
+       Expected: The file name does not have the .json file extension. Error details shown in the status message.
+
+   1. Test case: `export /`<br>
+      Expected: The file path is invalid as it is a directory. Error details shown in the status message.
+
+
+### Importing data file
+
+1. Exporting the data file.
```
Comment (Goh-Li-Ting) [inline docs/DeveloperGuide.md:826]: Should this be importing instead of exporting?

**Review by Goh-Li-Ting — COMMENTED** [review]
I think you can change the filtered label in the diagram to filtered and sorted.

Could you also help to remove this section of the developer guide?
image

**Comment by aexolate** [comment]
> I think you can change the filtered label in the diagram to filtered and sorted.
> 
> Could you also help to remove this section of the developer guide? image

Thanks, I have made the changes.

**Review by Weiennn — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -761,3 +799,37 @@ testers are expected to do more *exploratory* testing.
 
     1. Test case: `history 0`<br>
        Expected: Error details shown in the status message.
+
+### Exporting data file
+
+1. Exporting the data file.
+
+    1. Test case: `export fy2324.json`<br>
+       Prerequisite: The application has write access to the directory where EmployeeManager is.
```
Comment (Goh-Li-Ting) [inline docs/DeveloperGuide.md:808]: Thanks for making the changes! I think there should be a `<br>` here too!

**Review by Goh-Li-Ting — APPROVED** [review]
LGTM

**Pull request merged**